### PR TITLE
Migrate test support files to TS

### PR DESCRIPTION
**Pull request by uladzimirdev**
Migrate support files to TS

pre-requisite for https://github.com/metabase/metabase/pull/36109

### CODE REVIEW

File: frontend/src/metabase-lib/expressions/__support__/expressions.ts [hunk 1]
```diff
@@ -26,7 +28,7 @@ const database = createMockDatabase({
   name: "db",
   tables: [
     createMockTable({
-      db: DB_ID,
+      db_id: DB_ID,
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/expressions/__support__/expressions.ts:31]: thanks TS

File: frontend/test/metabase/lib/expressions/formatter.unit.spec.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import { format } from "metabase-lib/expressions/format";
-
-import shared from "metabase-lib/expressions/__support__/shared";
+import { dataForFormatting } from "metabase-lib/expressions/__support__/shared";
 
 describe("metabase-lib/expressions/format", () => {
-  describe.each(shared)("%s", (name, cases, opts) => {
+  describe.each(dataForFormatting)("%s", (_name, cases, opts) => {
```
Comment (uladzimirdev) [inline frontend/test/metabase/lib/expressions/formatter.unit.spec.js:5]: `name` wasn't used

File: frontend/src/metabase-lib/expressions/__support__/expressions.ts [hunk 3]
```diff
@@ -107,19 +109,20 @@ const database = createMockDatabase({
     "foreign-keys",
     "native-parameters",
     "expressions",
-    "advanced-math-expressions",
+    // TODO: uladzimir. Either fix a type of remove this feature
+    // Type '"advanced-math-expressions"' is not assignable to type 'DatabaseFeature'
+    // "advanced-math-expressions",
```
Comment (kamilmielnik) [inline frontend/src/metabase-lib/expressions/__support__/expressions.ts:114]: I think we should update `DatabaseFeature`

**Review by kamilmielnik — APPROVED** [review]
:+1:

**Review by ranquild — APPROVED** [review]
Comment about DatabaseFeature

File: frontend/src/metabase-lib/expressions/__support__/expressions.ts (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase-lib/expressions/__support__/expressions.ts]: fixed https://github.com/metabase/metabase/pull/36265/commits/409d772b352acc598cb8e3372d555ab73dee9fc5

**Pull request merged**